include/rocksdb/cache.h

```diff
@@ -363,11 +352,29 @@ extern std::shared_ptr<Cache> NewClockCache(
     CacheMetadataChargePolicy metadata_charge_policy =
         kDefaultCacheMetadataChargePolicy);
 
+// A Cache maps keys to values resident in memory, tracks reference counts
+// on those key-value entries, and is able to remove unreferenced entries
+// whenever it wants. All operations are fully thread safe except as noted.
+// Inserted values have a specified "charge" which is some quantity in
+// unspecified units, typically bytes of memory used. A Cache will typically
+// have a finite capacity in units of charge, and evict entries as needed
+// to stay at or below that capacity.
+//
+// NOTE: This API is for expert use only and is more intended for providing
+// custom implementations than for calling into. It is subject to change
+// as RocksDB evolves, especially the RocksDB block cache.
 class Cache {
  public:  // opaque types
   // Opaque handle to an entry stored in the cache.
   struct Handle {};
```

Comment:
A `Handle` can be ready or waiting to be resolved, and the current way of determining that by looking at the value is hacky. Maybe as part of the refactoring, we can implement a cleaner way of doing that.

Reply:
I wanted to do that before, but it would take deeper changes to the read path than even I have done here. IIRC there are many levels of function calls where a Handle might be pending or not, and being more explicit would probably take more code and come with a measurable performance regression.